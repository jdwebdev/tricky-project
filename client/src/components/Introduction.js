import React from 'react';
import { useState } from 'react';
import data from '../data';
import PropTypes from 'prop-types';


const Introduction = (props) => {
    
    const phraseList = data.introText;
    const [currentPhrase, setCurrentPhrase] = useState(0);
    const [end, setEnd] = useState(false);

    const handleClick = () => {
        const current = currentPhrase + 1;
        if (current === phraseList.length -1) {
            setEnd(true);
        }
        setCurrentPhrase(current);
    }

    return (
        <div>
            <div className="introductionContainer">
                <p>{phraseList[currentPhrase]}</p>
            </div>
            {end === false &&             
                <button 
                    className="nextButton"
                    onClick={handleClick}
                >Suivant â–º</button>
            }
            {end &&
                <div className="startBtnContainer"> 
                    <button className="startBtn" onClick={props.onClick}>Commencer</button>
                </div>
            }
        </div>
    )
}


Introduction.propTypes = {
    onClick: PropTypes.func.isRequired
}

export default Introduction;