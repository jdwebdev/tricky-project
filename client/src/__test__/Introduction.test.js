/* eslint-disable no-undef */
import React from 'react'
import Introduction from '../components/Introduction'
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"


describe("Introduction component", () => {
  it("renders Introduction component without crashing", () => {
    shallow(<Introduction onClick={jest.fn()}/>)
  })

  let introWrapper;
  beforeEach(() => {
    introWrapper = shallow(<Introduction onClick={jest.fn()}/>)
  })

  it("renders correctly", () => {
    expect(shallowToJson(introWrapper)).toMatchSnapshot()
  })

  it('display first phrase correctly', () => {
    const p = introWrapper.find('p').text()
    expect(p).toEqual("Bienvenue dans Tricky Project")    
  })

  it('display next phrase on click', () => {
    const button = introWrapper.find('.nextButton')
    button.simulate("click")
    expect(introWrapper.find('p').text()).toEqual('Le but du jeu est simple : vous devez vous rendre Ã  la pharmacie afin de recevoir votre dose de vaccin.')
  })

})