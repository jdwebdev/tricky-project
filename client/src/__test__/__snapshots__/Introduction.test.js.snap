// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component renders correctly 1`] = `
<div
  className="introductionContainer"
>
  <p>
    Bienvenue dans Tricky Project
  </p>
  <button
    className="nextButton"
    onClick={[Function]}
  >
    Suivant â–º
  </button>
</div>
`;
